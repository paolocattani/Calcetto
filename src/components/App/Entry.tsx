import React, { StrictMode } from 'react';
import { PersistGate } from 'redux-persist/integration/react';
import { store, persistor } from '../../redux/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import { LoadingModal } from 'components/core/generic/Commons';

const Application: React.FC = () => (
  <Provider store={store}>
    <PersistGate loading={<LoadingModal show={true} />} persistor={persistor}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </PersistGate>
  </Provider>
);
const Entry: React.FC = () =>
  process.env.NODE_ENV === 'development' ? (
    <StrictMode>
      <Application />
    </StrictMode>
  ) : (
    <Application />
  );

export default Entry;
