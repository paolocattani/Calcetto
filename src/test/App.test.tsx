import React from 'react';
import { shallow } from 'enzyme';
import Entry from 'components/App/Entry';
import { shallowToJson } from 'enzyme-to-json';

describe('Application ', () => {
  it('should render correctly', () => {
    const component = shallow(<Entry />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
