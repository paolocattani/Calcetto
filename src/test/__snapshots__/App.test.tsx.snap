// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application  should render correctly 1`] = `
<StrictMode>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>
</StrictMode>
`;
